src/server/v2/resources/images.py

```diff
@@ -226,16 +229,38 @@ def patch(self, image_id):
                     if problem:
                         current_app.logger.info("%s Could not validate link artifact or artifact doesn't exist", log_id)
                         return problem
+                setattr(image, key, value)
             elif key == "arch":
                 current_app.logger.info(f"Patching architecture with {value}")
                 image.arch = value
+                setattr(image, key, value)
+            elif key == 'metadata':
+                # The API represents metadata keys as a dictionary, but the patchset is provided as a list of
+                # changes that need to be applied to the metadata itself.
+                for changeset in value:
+                    operation = changeset.get('operation')
+                    annotation_key = changeset.get('key')
+                    annotation_value = changeset.get('value', '')
+                    current_app.logger.debug("Image Patch changeset: Current: %s -> %s %s %s"
+                                            % (image.metadata, operation, annotation_key, annotation_value))
+                    if operation not in ['set', 'remove']:
+                        current_app.logger.info(f"Unknown requested operation change '{operation}'.")
+                        return generate_data_validation_failure(errors=[])
+                    current_app.logger.debug("Image Patch changeset: %s %s %s" % (operation, annotation_key, annotation_value))
+                    if operation == 'set':
+                        image.metadata[annotation_key] = annotation_value
+                    elif operation == 'remove':
+                        try:
+                            del image.metadata[annotation_key]
+                        except KeyError:
+                            current_app.logger.info("No-op when removing non-existent metadata from IMS record.")
+                    current_app.logger.debug("Image metadata result: %s" %(image.metadata))
             else:
                 current_app.logger.info(f"{log_id} Not able to patch record field {key} with value {value}")
                 return generate_data_validation_failure(errors=[])
-
-            setattr(image, key, value)
+        current_app.logger.info(f"{log_id} image metadata information dump: '%s'" % image.metadata)
```

Comment:
Question: is there a reason you didn't do: '... dump: '{image.metadata}'' instead of the % operator? Seems to be mixing string formatting styles.